internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -2448,56 +2451,222 @@ func TestComputeDesiredMachine(t *testing.T) {
 	expectedUpdatedMachine.Spec.InfrastructureRef = *existingMachine.Spec.InfrastructureRef.DeepCopy()
 	expectedUpdatedMachine.Spec.Bootstrap.ConfigRef = existingMachine.Spec.Bootstrap.ConfigRef.DeepCopy()
 
-	tests := []struct {
-		name            string
-		existingMachine *clusterv1.Machine
-		want            *clusterv1.Machine
-	}{
+	tests := []computeDesiredMachineTestCase{
+		{
+			name: "should return the correct Machine object when creating a new Machine",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					MachineNamingStrategy: &clusterv1.MachineNamingStrategy{
+						Template: "{{ .machineSet.name }}" + namingTemplateKey + "-{{ .random }}",
+					},
+					Template: machineTemplateSpec,
+				},
+			},
+			existingMachine: nil,
+			wantMachine:     expectedNewMachine,
+			wantName: []gomegatypes.GomegaMatcher{
+				HavePrefix(msName + namingTemplateKey),
+				Not(HaveSuffix("00000")),
```

Comment:
Isn't `00000` a valid generated random ending? (although very unlikely?)


```suggestion
				Not(HaveSuffix("-")),
```

Reply:
done